controllers/__init__.py

```diff
@@ -47,14 +48,25 @@ def bootstrap_indexes(self) -> None:
                     f"Created DB index {keys}, collection: {collection}"
                 )
 
-    def get_circuits(self, archived: Optional[bool] = False) -> Dict:
+    def get_circuits(self, archived: Optional[bool] = False,
+                     metadata: dict = None) -> Dict:
         """Get all circuits from database."""
         aggregation = []
+        options = {"null": None, "true": True, "false": False}
         match_filters = {"$match": {}}
+        aggregation.append(match_filters)
         if archived is not None:
+            archived = options.get(archived, False)
             match_filters["$match"]["archived"] = archived
-            aggregation.append(match_filters)
-
+        if metadata:
+            for key in metadata:
+                if "metadata." in key[:9]:
+                    try:
+                        match_filters["$match"][key] = int(metadata[key])
+                    except ValueError:
+                        item = metadata[key].lower()
```

Comment:
`lower()` is great to normalize the `option` values, but since metadata values can be filtered by `str`, if the evc has this metadata `telemetry: { enabled: true, switch: 1, city: 'Curitiba' },`, then `city` would never be able to be filtered since it'd be normalize as `'curitiba'`, so maybe if the lowered `item` isn't on `options` then we should pass `metadata[key]` as it is?

Reply:
You are right, changed.